src/Network/Request.php

```diff
@@ -1359,4 +1360,45 @@ public function offsetUnset($name)
     {
         unset($this->params[$name]);
     }
+
+    /**
+     * Returns string representation of the current request in pluginDot syntax
+     *
+     * @params array $options Valid keys are `prefix`, `action`, `pass`, and `query`
+     * @return string Path
+     */
+    public function toPath($options = []) {
+        $prefix = $plugin = $controller = $action = $pass = $query = null;
+        if ($this->plugin) {
+            $plugin = $this->plugin . '.';
+        }
+        if ($prefix = $this->param('prefix')) {
+            $controller = Inflector::camelize($prefix) . '/';
+        }
+        $controller .= $this->param('controller') . 'Controller';
+
+        if (!empty($options['action'])) {
+            $action = '/' . $this->param('action');
+        }
+
+        if (!empty($options['pass']) && $passedArgs = $this->param('pass')) {
+            $pass = '/' . implode('/', $passedArgs);
+        }
+
+        if (!empty($options['query']) && $this->query) {
+            $query = '?' . http_build_query($this->query);
+        }
+
+        return $plugin . $controller . $action . $pass . $query;
```

Comment:
`toPath()` implies URLs to me. This looks like a URL but is not one at the same time. What kinds of uses do you see for this result?